### ISSUE TITLE
Typescript generation of collections nested inside maps.

### ISSUE THREAD

**Issue by anticwitee** [vaadin/fusion#109]
### Description of the bug 
Typescript generation of Java types can seemingly not handle nested collections or arrays inside a map.

Given the following field;
```
public Map<String, List<String>> nestedList;
```

In Vaadin 20.X the generated type for a field above looks like;
```
  get nestedList(): ArrayModel<string, StringModel> {
    return this[_getPropertyModel]('nestedList', ArrayModel, [false, StringModel, [false]]);
  }
```

In Vaadin 21.X there is an extraneous `>` which is placed in between `StringModel`. This unfortunately causes the frontend compilation to fail.
```
get nestedList(): ArrayModel<string, String>Model> {
    return this[_getPropertyModel]('nestedList', ArrayModel, [false, String>Model, [false]]);
  }
```

### Minimal reproducible example
I downloaded a Vaadin 21 Fusion project from Vaadin Start, then chose the "Person Form" view as it has Java endpoints.
I added a new endpoint - returning a type that has a nested list inside a map.
Adding this endpoint to a MobX store and invoking it from a web component is sufficient to trigger the build error.

An example project is available [here](https://github.com/anticwitee/vaadin-fusion-collections-bug-report)

### Expected behavior
Vaadin should generate a usable typescript type for Java classes with fields of this form.

### Actual behavior
Webpack fails to compile due to a parsing error.

[error-log.txt](https://github.com/vaadin/fusion/files/7177510/error-log.txt)

### Versions:
    - Vaadin / Fusion version: 21.0.1
    - Node version:  12.18.3
    - Java version: 11.0.12
    - OS version: Linux 5.10.61-1-Manjaro x86_64

**Comment by Lodin** [comment]
Duplicate of #51.
Solved by vaadin/flow#11522

**Comment by haijian-vaadin** [comment]
Thanks for reporting the issue, we are currently cherry-picking https://github.com/vaadin/flow/pull/11522 to Flow 8/Vaadin 21. The problem should be fixed once the cherry-pick is done.

**Comment by haijian-vaadin** [comment]
This is fixed in https://github.com/vaadin/flow/pull/11901. A new release could be expected next week.
Meanwhile, the generated type for `public Map<String, List<String>> nestedList;` becomes 
```
get nestedList(): ObjectModel<Record<string, ModelValue<ArrayModel<string, StringModel>>>> {
    return this[_getPropertyModel]('nestedList', ObjectModel, [true]);
}
```
with the fix, which is more correct.
